### ISSUE TITLE
False positive: "Objects are always passed by reference" with iterable typehint

### ISSUE THREAD

**Issue by DemoniacDeath** [kalessil/phpinspectionsea#1531]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.3 |
| Language level | PHP 7.4 |
| Inspection | [Referencing objects](https://github.com/kalessil/phpinspectionsea/blob/master/docs/confusing-constructs.md#referencing-objects) |
| Inspection message | `Objects are always passed by reference` |

Current behaviour
---
In the following case
```php
<?php

function test(iterable &$test) {

}
```
the `iterable &$test` is marked with inspection `Objects are always passed by reference`

Expected behaviour
---
an argument typhinted as `iterable` should not be considered exclusively object. it can be an array, so this inspection is false positive and should not be triggered in such a case.

Environment details
---
PhpStorm 2019.3.3
Build #PS-193.6494.47, built on February 12, 2020
Licensed to Mykhailo Skvortsov
Subscription is active until September 13, 2020
Runtime version: 1.8.0_202-b1483.37 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 5.4.21-1-lts
GC: ParNew, ConcurrentMarkSweep
Memory: 1925M
Cores: 8
Registry: run.processes.with.pty=TRUE, debugger.watches.in.variables=false, ide.balloon.shadow.size=0, focus.follows.mouse.workarounds=true
Non-Bundled Plugins: com.intellij.properties, com.alayouni.ansiHighlight, com.intellij.ideolog, com.intellij.kubernetes, com.kalessil.phpStorm.phpInspectionsEA, de.espend.idea.php.annotation, fr.adrienbrault.idea.symfony2plugin, ru.adelf.idea.dotenv

**Comment by ea-inspections-team** [comment]
Thanks for reporting @DemoniacDeath! Fixed!
